gossip/handler.go

```diff
@@ -0,0 +1,72 @@
+// Copyright (C) 2019-2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package gossip
+
+import (
+	"context"
+	"time"
+
+	bloomfilter "github.com/holiman/bloomfilter/v2"
+
+	"github.com/ava-labs/avalanchego/codec"
+	"github.com/ava-labs/avalanchego/ids"
+)
+
+// var _ p2p.Handler = &Handler[any, *any]{}
+
+func NewHandler[T any, U GossipableAny[T]](set Set[U], codec codec.Manager, codecVersion uint16) *Handler[T, U] {
+	return &Handler[T, U]{
+		set:          set,
+		codec:        codec,
+		codecVersion: codecVersion,
+	}
+}
+
+type Handler[T any, U GossipableAny[T]] struct {
+	set          Set[U]
+	codec        codec.Manager
+	codecVersion uint16
+}
+
+func (h Handler[T, U]) AppGossip(context.Context, ids.NodeID, []byte) error {
```

Comment:
Should we embed a no-op handler so that this only needs to override the functions that we want to override?

Reply:
We probably wanna implement the NoOp handler in the sdk package to re-use it, so I think for now we can leave this unimplemented and we can implement a NoOp handler as a follow-up. https://github.com/ava-labs/avalanchego/issues/1814